src/discussions/comments/CommentsView.test.jsx

```diff
@@ -142,58 +173,105 @@ describe('CommentsView', () => {
+    mockAxiosReturnPagedCommentsResponses();
   });
 
-  // TODO: use test id to prevent breaking from text changes
-  const findLoadMoreCommentsButton = () => screen.findByRole('button', { name: /load more comments/i });
+  const findLoadMoreCommentsButton = () => (
+    screen.findByTestId('load-more-comments')
+  );
+
+  const findLoadMoreCommentsResponsesButton = () => (
+    screen.findByTestId('load-more-comments-responses')
+  );
 
   it('initially loads only the first page', async () => {
     const firstPageComment = mockCommentsPaged[0][0];
     const secondPageComment = mockCommentsPaged[1][0];
-    mockAxiosReturnPagedComments();
     renderComponent();
 
-    await screen.findByText(firstPageComment.renderedBody);
-    expect(screen.queryByText(secondPageComment.renderedBody)).not.toBeInTheDocument();
+    await screen.findByText(firstPageComment.rendered_body);
+    expect(screen.queryByText(secondPageComment.rendered_body)).not.toBeInTheDocument();
   });
 
   it('pressing load more button will load next page of comments', async () => {
     const secondPageComment = mockCommentsPaged[1][0];
-    mockAxiosReturnPagedComments();
     renderComponent();
 
     const loadMoreButton = await findLoadMoreCommentsButton();
     fireEvent.click(loadMoreButton);
 
-    await screen.findByText(secondPageComment.renderedBody);
+    await screen.findByText(secondPageComment.rendered_body);
   });
 
   it('newly loaded comments are appended to the old ones', async () => {
     const firstPageComment = mockCommentsPaged[0][0];
     const secondPageComment = mockCommentsPaged[1][0];
-    mockAxiosReturnPagedComments();
     renderComponent();
 
     const loadMoreButton = await findLoadMoreCommentsButton();
     fireEvent.click(loadMoreButton);
 
-    await screen.findByText(secondPageComment.renderedBody);
+    await screen.findByText(secondPageComment.rendered_body);
     // check that comments from the first pages are also displayed
-    expect(screen.queryByText(firstPageComment.renderedBody)).toBeInTheDocument();
+    expect(screen.queryByText(firstPageComment.rendered_body)).toBeInTheDocument();
   });
 
   it('load more button is hidden when no more comments pages to load', async () => {
     const totalePages = mockCommentsPaged.length;
     const lastPageComment = mockCommentsPaged[totalePages - 1][0];
-    mockAxiosReturnPagedComments();
     renderComponent();
 
     const loadMoreButton = await findLoadMoreCommentsButton();
     for (let page = 1; page < totalePages; page++) {
       fireEvent.click(loadMoreButton);
     }
 
-    await screen.findByText(lastPageComment.renderedBody);
+    await screen.findByText(lastPageComment.rendered_body);
     await expect(findLoadMoreCommentsButton()).rejects.toThrow();
   });
+
+  describe('comments responses', () => {
+    it('initially loads only the first page', async () => {
+      renderComponent();
+
+      await screen.findByText('comment number 1', { exact: false });
```

Comment:
I think this should be wrapped in an expect as well.

Reply:
@xitij2000 Can you please explain why?
If the element has been found, expect doesn't add anything.
If the element hasn't been found, an error would be thrown, which would lead to the test failing and the correct message would be displayed, and expect is not needed for that.
If there was an element, let's say a loader, which I could wait for to disappear, it would be easy to change `findByText` to `queryByText` and wrap it with a expect. If you think it's worth it, I can change the code to wait for `store.comments.status` to change to `RequestStatus.SUCCESSFUL` and then do the assertion, but my understanding was that it's bad to rely on internal state, and instead try to rely on UI.

Reply:
Fair enough!